src/pymor/operators/constructions.py

```diff
@@ -1321,6 +1325,61 @@ def as_source_array(self, mu=None):
         return self.operators[operator_number].as_source_array(mu=mu)
 
 
+def vector_array_to_selection_operator(array, time_instances=None, initial_time=None, end_time=None, name=None):
+    """Create a |SelectionOperator| from a |VectorArray|.
+
+    A |SelectionOperator| is created that selects an element from the |VectorArray| `array`
+    depending on the time component of the parameter that is passed to the |Operator|.
+    If `time_instances` is provided, these values are used for selection according to the
+    behavior of the |SelectionOperator|.
+    If `initial_time` and `end_time` are used, the time interval is split equidistantly where
+    the number of subintervals is determined by the length of `array`.
+    In time, the operator is therefore piecewise constant and the discontinuity points are
+    at `t_n + eps`, i.e.::
+
+        -infty ------- initial_time ----------- t_1 ----------- t_2 ----------- ... ----------- t_{n-1} ----------- end_time ------------------------ infty
+                                     |               |               |                                   |                    |
+        ------------ array[0] -------|--- array[1] --|--- array[2] --|--------- ... ---------------------|---- array[n-2] ----|-------- array[n-1] --------
+                                     |               |               |                                   |                    |
```

Comment:
Is this image correct? If I look at the code, I think it would be `array[n-1]` after `t_{n-1} + eps`?

Reply:
(I would have expected it to be the other way around: `array[0]` until `t_1 - eps`. But I don't know if one is unconditionally better than the other. Probably depends on the method that is analyzed.)

Reply:
> Is this image correct? If I look at the code, I think it would be `array[n-1]` after `t_{n-1} + eps`?

Yes, you're right.

Reply:
> (I would have expected it to be the other way around: `array[0]` until `t_1 - eps`. But I don't know if one is unconditionally better than the other. Probably depends on the method that is analyzed.)

Honestly, this way, it seemed easier for me to implement. I also have no idea which version is better.